src/components/Carousel/Carousel.js

```diff
@@ -0,0 +1,159 @@
+import React, { Component } from 'react';
+import cx from 'classnames';
+
+import PropTypes from 'prop-types';
+import scrollTo from './scrollToAnimate';
+
+import './Carousel.scss';
+export const CN = 'carousel';
+
+class Carousel extends Component {
+  constructor(props) {
+    super(props);
+    this.carouselViewport = React.createRef();
+    this.handleClick = this.handleClick.bind(this);
+    this.onResize = this.onResize.bind(this);
+    this.onScroll = this.onScroll.bind(this);
+    this.state = {
+      numOfSlidesToScroll: 3,
+      allTheWayLeft: false,
+      allTheWayRight: false,
+    };
+  }
+
+  componentDidMount() {
+    const { scrollWidth } = this.carouselViewport.current;
+    const { length } = this.props.children;
+
+    this.setState({ widthOfSlide: scrollWidth / length },
+      () => {
+        this.resizeTheCarousel();
+        this.checkNumOfSlidesToScroll();
+        this.checkIfSlidesAllTheWayOver();
+        window.addEventListener('resize', this.onResize);
+      });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.onResize);
+  }
+
+  onScroll() {
+    this.checkIfSlidesAllTheWayOver();
+  }
+
+  onResize() {
+    this.resizeTheCarousel();
+    this.checkNumOfSlidesToScroll();
+  }
+
+  resizeTheCarousel() {
+    const { widthOfSlide } = this.state;
+    let carouselWidth = '0';
+
+    if (window.innerWidth > 1280) {
+      carouselWidth = `${widthOfSlide * 5}px`;
+    } else if (window.innerWidth > 1024) {
+      carouselWidth = `${widthOfSlide * 4}px`;
+    } else if (window.innerWidth > 768) {
+      carouselWidth = `${widthOfSlide * 3}px`;
+    } else {
+      carouselWidth = `${widthOfSlide * 2}px`;
+    }
+
+    this.carouselViewport.current.parentElement.style.width = carouselWidth;
+  }
+
+  checkIfSlidesAllTheWayOver() {
+    // if scrollLeft == 0
+    // hide left button
+    const allTHeWayLeftValue = (this.carouselViewport.current.scrollLeft === 0);
+
+    // if scrollLeft + viewPortOffset.length === whole viewPort length
+    // 9 cards - each 120px: 9 * 210all === whole viewPort length
+    // hide the rightScrollButton
+    const { scrollLeft, clientWidth, scrollWidth } = this.carouselViewport.current;
```

Comment:
move it to top and use `scrollLeft` for allTHeWayLeftValue

Reply:
done